--- origsrc/netpbm/lib/libsystem.c	2010-02-10 20:50:24.000000000 -0600
+++ src/netpbm/lib/libsystem.c	2010-02-10 20:54:50.796071200 -0600
@@ -246,8 +246,10 @@ signalName(unsigned int const signalClas
             return "SIGWINCH";
         case SIGIO:
             return "SIGIO";
+#ifdef SIGPWR
         case SIGPWR:
             return "SIGPWR";
+#endif
         case SIGSYS:
             return "SIGSYS";
         default:
