--- origsrc/netpbm/Makefile.config	1969-12-31 18:00:00.000000000 -0600
+++ src/netpbm/Makefile.config	2008-04-23 17:18:13.609375000 -0500
@@ -0,0 +1,71 @@
+#
+# Makefile.config for Cygwin
+#
+DEFAULT_TARGET = nonmerge
+BUILD_FIASCO = Y
+CC = gcc
+LD = $(CC)
+LINKERISCOMPILER=Y
+LINKER_CAN_DO_EXPLICIT_LIBRARY=Y
+CC_FOR_BUILD = $(CC)
+LD_FOR_BUILD = $(LD)
+INSTALL = install
+STRIPFLAG = 
+SYMLINK = ln -s
+MANPAGE_FORMAT = nroff
+AR = ar
+RANLIB = ranlib
+LEX = flex
+CFLAGS = -O3 -pipe -ffast-math -pedantic \
+          -Wall -Wno-uninitialized -Wmissing-declarations
+EXE = .exe
+LDFLAGS =
+LDSHLIB = -shared -Wl,--enable-auto-image-base
+LDRELOC = ld -r
+CFLAGS_SHLIB = -DPIC
+SHLIB_CLIB =
+NEED_RUNTIME_PATH = N
+RPATHOPTNAME = -rpath
+#NETPBMLIB_RUNTIME_PATH = /usr/lib/netpbm
+#
+# system libraries
+#
+TIFFLIB = /usr/lib/libtiff.dll.a
+JPEGLIB = /usr/lib/libjpeg.dll.a
+PNGLIB = /usr/lib/libpng12.dll.a
+ZLIB = /usr/lib/libz.dll.a
+JBIGLIB = /usr/lib/libjbig.dll.a
+#
+# jpeg2ktopam is incompatible with the system jasper; use internal instead
+#
+JASPERLIB = $(INTERNAL_JASPERLIB)
+JASPERHDR_DIR = $(INTERNAL_JASPERHDR_DIR)
+URTLIB = $(BUILDDIR)/urt/librle.a
+URTHDR_DIR = $(SRCDIR)/urt
+X11LIB = /usr/lib/libX11.dll.a
+X11HDR_DIR =
+LINUXSVGALIB = NONE
+OMIT_NETWORK =
+#NETWORKLD = -lresolv
+VMS = 
+#
+# Install options
+#
+PKGDIR_DEFAULT = /tmp/netpbm
+INSTALL_PERM_BIN =  755       # u=rwx,go=rx
+INSTALL_PERM_LIBD = 755       # u=rwx,go=rx
+INSTALL_PERM_LIBS = 644       # u=rw,go=r
+INSTALL_PERM_HDR =  644       # u=rw,go=r
+INSTALL_PERM_MAN =  644       # u=rw,go=r
+INSTALL_PERM_DATA = 644       # u=rw,go=r
+SUFFIXMANUALS1 = 1
+SUFFIXMANUALS3 = 3
+SUFFIXMANUALS5 = 5
+NETPBMLIBTYPE = dll
+NETPBMLIBSUFFIX = dll
+STATICLIB_TOO = y
+STATICLIBSUFFIX = a
+SHLIBPREFIXLIST = cyg lib
+NETPBMSHLIBPREFIX = $(firstword $(SHLIBPREFIXLIST))
+DLLVER = -$(NETPBM_MAJOR_RELEASE)
+NETPBM_DOCURL = http://netpbm.sourceforge.net/doc/
--- origsrc/netpbm/converter/other/Makefile	2008-04-23 17:17:20.000000000 -0500
+++ src/netpbm/converter/other/Makefile	2008-04-23 17:59:35.265625000 -0500
@@ -152,7 +152,7 @@
 ifeq ($(shell libpng-config --version),)
   PNGLIB_LIBOPTS = $(shell $(LIBOPT) $(LIBOPTR) $(PNGLIB) $(ZLIB))
 else
-  PNGLIB_LIBOPTS = $(shell libpng-config --ldflags)
+  PNGLIB_LIBOPTS = $(shell libpng-config --ldflags) $(ZLIB)
 endif
 
 pngtopnm: %: %.o $(NETPBMLIB) $(LIBOPT)
@@ -216,7 +216,7 @@
 # Remember that $(SYMLINK) might just be a copy command.
 # backward compatibility: program used to be named pnmnoraw
 	cd $(PKGDIR)/bin ; \
-	$(SYMLINK) pnmtoplainpnm$(EXE) pnmnoraw
+	$(SYMLINK) pnmtoplainpnm pnmnoraw
 # backward compatibility: program used to be gemtopbm
 	cd $(PKGDIR)/bin ; \
 	$(SYMLINK) gemtopnm$(EXE) gemtopbm
--- origsrc/netpbm/converter/other/jpeg2000/libjasper/Makefile.common	2008-04-23 17:17:06.000000000 -0500
+++ src/netpbm/converter/other/jpeg2000/libjasper/Makefile.common	2008-04-23 17:18:13.656250000 -0500
@@ -23,7 +23,7 @@
 
 INCLUDES = -I$(JASPERSRCDIR)/include -Iimportinc
 
-DEFS = -DHAVE_LIBM=1 -DSTDC_HEADERS=1 -DHAVE_FCNTL_H=1 -DHAVE_LIMITS_H=1 -DHAVE_UNISTD_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STDDEF_H=1 -DEXCLUDE_BMP_SUPPORT -DEXCLUDE_RAS_SUPPORT -DEXCLUDE_MIF_SUPPORT -DEXCLUDE_JPG_SUPPORT -DEXCLUDE_PGX_SUPPORT -DEXCLUDE_PNM_SUPPORT
+DEFS = -DHAVE_LIBM=1 -DSTDC_HEADERS=1 -DHAVE_FCNTL_H=1 -DHAVE_LIMITS_H=1 -DHAVE_UNISTD_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STDDEF_H=1 -DHAVE_STDINT_H=1 -DEXCLUDE_BMP_SUPPORT -DEXCLUDE_RAS_SUPPORT -DEXCLUDE_MIF_SUPPORT -DEXCLUDE_JPG_SUPPORT -DEXCLUDE_PGX_SUPPORT -DEXCLUDE_PNM_SUPPORT
 
 $(LIB_OBJECTS):%.o:%.c
 	$(CC) -c $(CPPFLAGS) $(CFLAGS) $(INCLUDES) $(DEFS) $(CADD) $<
--- origsrc/netpbm/converter/other/pamtodjvurle.c	2008-04-23 17:17:21.000000000 -0500
+++ src/netpbm/converter/other/pamtodjvurle.c	2008-04-23 17:18:13.671875000 -0500
@@ -179,7 +179,7 @@
   
   'transcolor' is a 3-deep tuple with the same maxval as the image.
 -----------------------------------------------------------------------------*/
-    uint32n rlevalue;         /* RLE-encoded color/valuex */
+    uint32_t rlevalue;         /* RLE-encoded color/valuex */
     int index;
 
     if (count > 0) {
--- origsrc/netpbm/converter/ppm/xpmtoppm.c	2008-04-23 17:17:34.000000000 -0500
+++ src/netpbm/converter/ppm/xpmtoppm.c	2008-04-23 17:18:13.671875000 -0500
@@ -112,6 +112,8 @@
 }
 
 
+#define getline _getline
+
 static char lastInputLine[MAX_LINE+1];
     /* contents of line most recently read from input */
 static bool backup;
@@ -120,7 +122,7 @@
     */
 
 
 static void
-getline(char * const line,
+_getline(char * const line,
         size_t const size,
         FILE * const stream) {
 /*----------------------------------------------------------------------------
--- origsrc/netpbm/converter/ppm/xvminitoppm.c	2008-04-23 17:17:34.000000000 -0500
+++ src/netpbm/converter/ppm/xvminitoppm.c	2008-04-23 17:18:13.687500000 -0500
@@ -53,7 +53,7 @@
 
 
 static void
-getline(FILE * const ifP,
+pm_getline(FILE * const ifP,
         char * const buf,
         size_t const size) {
 
@@ -68,7 +68,7 @@
             pm_error("unexpected EOF");
     }
 }
-
+#define getline pm_getline
 
 
 static void
--- origsrc/netpbm/lib/Makefile	2008-04-23 17:16:59.000000000 -0500
+++ src/netpbm/lib/Makefile	2008-04-23 17:18:13.687500000 -0500
@@ -133,7 +133,6 @@
 $(NETPBMSHLIBPREFIX)netpbm$(DLLVER).dll: $(LIBOBJECTS) $(LIBOBJECTS_X)
 endif
 	$(LD) $(LDSHLIB) -Wl,--export-all-symbols \
-            -Wl,-soname,$(NETPBMSHLIBPREFIX)netpbm$(DLLVER).dll \
 	    -Wl,--output-def,$(NETPBMSHLIBPREFIX)netpbm$(DLLVER).def \
             -Wl,--out-implib,libnetpbm.dll.a -o $@ $(LDFLAGS) \
             $(LIBOBJECTS) $(LIBOBJECTS_X) $(LDLIBS) $(LADD) 
--- origsrc/netpbm/pm_config.in.h	2008-04-23 17:17:44.000000000 -0500
+++ src/netpbm/pm_config.in.h	2008-04-23 17:18:13.703125000 -0500
@@ -49,7 +49,7 @@
 
    The problematic C libraries treat _XOPEN_SOURCE the same way.
 */
-#if defined(__OpenBSD__) || defined (__NetBSD__) || defined(__bsdi__) || defined(__APPLE__)
+#if defined(__OpenBSD__) || defined (__NetBSD__) || defined(__bsdi__) || defined(__APPLE__) || defined(__CYGWIN__)
 #define POSIX_IS_IMPLIED
 #endif
 
@@ -116,8 +116,13 @@
    But we're going to wait a while in case our uint32_t strategy doesn't
    work.  04.08.24.
 */
+#ifdef __CYGWIN__
+typedef unsigned int uint32n;
+typedef signed int int32n;
+#else
 typedef uint32_t uint32n;
 typedef int32_t int32n;
+#endif
 
 /* CONFIGURE: On most BSD systems, malloc() gets declared in stdlib.h, on
 ** system V, it gets declared in malloc.h. On some systems, malloc.h
